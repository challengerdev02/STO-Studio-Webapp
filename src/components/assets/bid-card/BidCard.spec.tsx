import { cleanup, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BidCard } from './index';
import React from 'react';

describe('BidCard component', function () {
  afterEach(cleanup);
  let props = {
    user: '0x658d224566...F0cC',
    cryptoValue: '1.46 ETH',
    priceEquivalent: '$2,764.89',
    startDate: '343604800',
    endDate: '604800',
    userAvatar: '../../../public/verified-account-avatar.svg',
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<BidCard {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should check that the BidCard component displays', () => {
    const { getByTestId } = setup();

    expect(getByTestId('bid-card')).toBeInTheDocument();
  });

  it('should test the BidCard component has a user avatar', () => {
    const { getByTestId } = setup();
    const image = getByTestId('user-avatar');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', props.userAvatar);
  });

  it('should test the BidCard component has the user avatar displayed twice', () => {
    const { getByTestId } = setup();
    const secondImage = getByTestId('user-avatar-2');
    expect(secondImage).toBeInTheDocument();
    expect(secondImage).toHaveAttribute('src', props.userAvatar);
  });

  it('should test the BidCard component has the user blockchain address displayed on it.', () => {
    const { getByTestId } = setup();
    expect(getByTestId('block-chain-address')).toBeInTheDocument();
  });

  it('should test the bid on the BidCard component has a price', () => {
    const { getByTestId } = setup();
    const cryptoValue = getByTestId('crypto-value');
    expect(cryptoValue).toBeInTheDocument();
    expect(cryptoValue).toHaveTextContent(props.cryptoValue);
  });

  it('should test the BidCard component has a price equivalent.', () => {
    const { getByTestId } = setup();
    const priceEquivalent = getByTestId('price-equivalent');
    expect(priceEquivalent).toBeInTheDocument();
    expect(priceEquivalent).toHaveTextContent(props.priceEquivalent);
  });

  it('should test the BidCard component has the TimeBlock components painted on screen', () => {
    const { getByTestId } = setup();
    expect(getByTestId('days-timeblock')).toBeInTheDocument();
    expect(getByTestId('days-timeblock')).toHaveTextContent;

    expect(getByTestId('hours-timeblock')).toBeInTheDocument();
    expect(getByTestId('hours-timeblock')).toHaveTextContent;

    expect(getByTestId('minutes-timeblock')).toBeInTheDocument();
    expect(getByTestId('minutes-timeblock')).toHaveTextContent;

    expect(getByTestId('seconds-timeblock')).toBeInTheDocument();
    expect(getByTestId('seconds-timeblock')).toHaveTextContent;
  });

  it('should test the Place a bid button is painted on screen ', () => {
    const { getByRole } = setup();
    const button = getByRole('button', { name: /place a bid/i });
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
  });
});
