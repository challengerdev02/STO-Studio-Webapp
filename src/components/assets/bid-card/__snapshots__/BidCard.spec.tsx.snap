// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidCard component should match with snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered sc-bdvvtL bxbWpv"
    data-testid="bid-card"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="sc-gsDKAQ irBxls"
      >
        <div
          class="sc-dkPtRN gOyVxi"
        >
          <div
            class="sc-hKwDye bXwPHc"
          >
            <div
              class="sc-eCImPb iVIGGu"
            >
               
              <img
                alt=""
                class="sc-ksdxgE hCdvzg"
                data-testid="user-avatar"
                height="28px"
                src="../../../public/verified-account-avatar.svg"
                width="26px"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-dkPtRN gOyVxi"
        >
          <div
            class="sc-jRQBWg iXDZyk"
          >
            <span
              class="sc-iCfMLu eNRNSg"
            >
              Highest bid by
            </span>
             
            <span
              class="sc-furwcr heaATz"
              data-testid="block-chain-address"
              title="0x658d224566...F0cC"
            >
              0x658d224566...F0cC
            </span>
          </div>
          <div
            class="sc-gKclnd jhPXGB"
          >
            <span
              class="sc-pVTFL gwavzj"
              data-testid="crypto-value"
            >
              1.46 ETH
            </span>
            <span
              class="sc-jrQzAO dewBhk"
              data-testid="price-equivalent"
            >
              $2,764.89
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal sc-kDTinF kQesqT"
        role="separator"
      />
      <div
        class="sc-gsDKAQ irBxls"
      >
        <div
          class="sc-dkPtRN gOyVxi"
        >
          <div
            class="sc-hKwDye bXwPHc"
          >
            <div
              class="sc-eCImPb iVIGGu"
            >
               
              <img
                alt=""
                class="sc-ksdxgE hCdvzg"
                data-testid="user-avatar-2"
                height="28px"
                src="../../../public/verified-account-avatar.svg"
                width="26px"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-dkPtRN gOyVxi"
        >
          <div
            class="sc-jRQBWg iXDZyk"
          >
            <span
              class="sc-iCfMLu eNRNSg"
            >
              Auction ends in
            </span>
          </div>
          <div
            class="sc-gKclnd jhPXGB"
          >
            <div
              class="sc-crHmcD lgANCv"
            >
              <span
                class="sc-egiyK ilBKaU"
                data-testid="days-timeblock"
              />
               
              <span
                class="sc-bqiRlB JQzJU"
              >
                Days
              </span>
               
            </div>
            <div
              class="sc-crHmcD lgANCv"
            >
              <span
                class="sc-egiyK ilBKaU"
                data-testid="hours-timeblock"
              />
               
              <span
                class="sc-bqiRlB JQzJU"
              >
                Hours
              </span>
               
            </div>
            <div
              class="sc-crHmcD lgANCv"
            >
              <span
                class="sc-egiyK ilBKaU"
                data-testid="minutes-timeblock"
              />
               
              <span
                class="sc-bqiRlB JQzJU"
              >
                Minutes
              </span>
               
            </div>
            <div
              class="sc-crHmcD lgANCv"
            >
              <span
                class="sc-egiyK ilBKaU"
                data-testid="seconds-timeblock"
              />
               
              <span
                class="sc-bqiRlB JQzJU"
              >
                Seconds
              </span>
               
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-primary ant-btn-block sc-iqseJM cBDWUJ"
        type="button"
      >
        <span>
          Place a bid
        </span>
      </button>
    </div>
  </div>
</div>
`;
